### DISCUSSION TITLE
Captcha problem

### DISCUSSION THREAD

**Discussion by nanobapt** [nfrastack/container-lemonldap#52] [Q&A]
Hi,
Enabling registration make captcha that do not appear in the form. It happen that the font helvetica is missing.

Installing ghostscript-fonts solved the problem. Could you include the package in the image ?

Thnx

Baptiste

**Comment by tiredofit** [comment]
Sure thing - Tagged as the following:

`tiredofit/lemonldap:2.0-alpine-latest`

`tiredofit/lemonldap:2.0-latest`

`tiredofit/lemonldap:2.0-alpine-1.8.1`

`tiredofit/lemonldap:latest`

**Comment by nanobapt** [comment]
Thanks for that !
It seems that there is some issue with portal folder. Mail header refence logo inside htdocs folder. Which is not the case in this docker container.

Comment from user mailing list:

> We incorrecly make the assumption that the static dir containing
>the logo will always be at ../../htdocs/[and so on] relative to the template
>dir. This is the case in most LLNG installations, but not in the docker
>image you're using. You should fix the mail_header.tpl template until
>then, or arrange your docker image to follow the "standard" LLNG file
>hierarchy:
>$ ls /usr/share/lemonldap-ng/portal/*
>/usr/share/lemonldap-ng/portal/htdocs:
>index.fcgi  index.psgi  static
>/usr/share/lemonldap-ng/portal/templates:
>bootstrap  common

Can you do something for this ?

**Comment by tiredofit** [comment]
Saw that on the list. For some reason I went with that folder structure due to it being that way during early 2.0 builds when it was in Alpha when installing from packages. My 1.9 series images followed the htdocs folder structure.

I'm a bit nervous to change the paths at this time during the 2.x releases and would be more apt to change it when LLNG comes out with the 3.x series. Reason why is the sheer amount of users that would see breakage if they are referencing URLs or embedding custom files at runtime. With authentication service usually being the first visit for accessing resources on a network I'm sure we'd have some frustrated people rather quick, with no way to alert them ahead of time of the change.

Although not all is lost.  We can do a dirty hack here: `ln -s /usr/share/lemonldap-ng/portal /usr/share/lemonldap-ng/portal/htdocs`   Can you try this from inside the container and see if it resolves itself? If so I'll put this in as a temp fix.

**Comment by nanobapt** [comment]
I understand.
I tried with the symbolic link and it seems to work. This should be temporary workaround.

Thanks for your work !

**Comment by tiredofit** [comment]
Glad it works - Tagged as the following:

`tiredofit/lemonldap:2.0-alpine-latest`
`tiredofit/lemonldap:2.0-latest`
`tiredofit/lemonldap:2.0-alpine-1.8.2`
`tiredofit/lemonldap:latest`

**Comment by nanobapt** [comment]
Hi,
In new version there is some problems:
1 package s6-ipcserver & imagemagick-perlmagick
2 I've got an error with captcha module (again)

Magick.so doesn't found the symbol SortImagePixels.
Maybe something related to https://gitlab.ow2.org/lemonldap-ng/lemonldap-ng/-/issues/2053

Any idea ? (For now I've downgraded)

**Comment by tiredofit** [comment]
I'm going to pull the release from Docker Hub in the short term. I have a couple things on the go right now and may need a few days to look into this.
